app/src/org/commcare/fragments/connectId/ConnectIdPhoneVerificationFragmnet.java

```diff
@@ -170,9 +179,18 @@ public void afterTextChanged(Editable s) {
         return view;
     }
 
+    private void handleDeactivateButton() {
+        binding.connectDeactivateButton.setVisibility(deactivateButton == null ? View.GONE : View.VISIBLE);
+    }
+
+    private void handleKeyboardType() {
+        int inputType = (method == MethodUserDeactivate || method == MethodRecoveryAlternate) ? InputType.TYPE_CLASS_TEXT : InputType.TYPE_CLASS_PHONE;
```

Comment:
The text keyboard should only appear when the user is in the deactivation process since only the deactivation OTP is alphanumeric. Let me know if I've missed or forgotten anything.

Reply:
Done